src/store-api/src/logstore.rs

```diff
@@ -59,19 +66,37 @@ pub trait LogStore: Send + Sync + 'static + std::fmt::Debug {
     async fn list_namespaces(&self) -> Result<Vec<Self::Namespace>, Self::Error>;
 
     /// Create an entry of the associate Entry type
-    fn entry<D: AsRef<[u8]>>(&self, data: D, id: Id, ns: Self::Namespace) -> Self::Entry;
+    fn entry<D: AsRef<[u8]>>(&self, data: D, entry_id: EntryId, ns: Self::Namespace)
+        -> Self::Entry;
 
     /// Create a namespace of the associate Namespace type
     // TODO(sunng87): confusion with `create_namespace`
-    fn namespace(&self, id: namespace::Id) -> Self::Namespace;
+    fn namespace(
+        &self,
+        ns_id: NamespaceId,
+        wal_options: &WalOptions,
+    ) -> Result<Self::Namespace, Self::Error>;
```

Comment:
When does the `namespace` of kafka impl return an err?

Reply:
The interface is changed to not return a result since there're no cases that building namespace would incur an error.